src/D2L.CodeStyle.Analyzers/Diagnostics.cs

```diff
@@ -505,5 +505,15 @@ public static class Diagnostics {
 			description: "Unnecessarily listed in an analyzer whitelist."
 		 );
 
+		public static readonly DiagnosticDescriptor ExternalDataSetVisibilityProviderTypeParameterMatchesClass = new DiagnosticDescriptor(
+			id: "D2L0058",
+			title: "Type parameter of IExternalDataSetVisibilityProviderType must be the class using it.",
+			messageFormat: "The type parameter of '{0}' must be '{1}'.",

```

Comment:
_IExternalDataSetVisibilityProviderType_ is already in the title, I would get rid of the first parameter here